src/game/server/baseanimating.cpp

```diff
@@ -3598,6 +3605,25 @@ void CBaseAnimating::IgniteHitboxFireScale( float flHitboxFireScale )
 	pFlame->SetHitboxFireScale( flHitboxFireScale );
 }
 
+void CBaseAnimating::Extinguish() {
+	if ( !IsOnFire() )
+		return;
+
+	// get the flame effect
+	CEntityFlame *pFlame = dynamic_cast<CEntityFlame*>( GetEffectEntity() );
+	if (pFlame)
+	{
+		pFlame->SetThink(&CBaseEntity::SUB_Remove);
+		pFlame->SetNextThink(gpGlobals->curtime + 0.1f);
+	}
+
+	// remove the OnFire flag
+	RemoveFlag(FL_ONFIRE);
+
+	// fire OnExtinguish output
+	m_OnExtinguish.FireOutput(this, this);
```

Comment:
Follow Valve's styling convention (Allman-style braces, spaces inbetween parentheses)

Reply:
Resolved in 6fb4501